core/src/main/java/org/infinispan/remoting/inboundhandler/TopologyMode.java

```diff
@@ -0,0 +1,70 @@
+package org.infinispan.remoting.inboundhandler;
+
+import org.infinispan.statetransfer.StateTransferLock;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * It checks or waits until the required topology is installed.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public enum TopologyMode {
+   WAIT_TOPOLOGY {
+      @Override
+      void await(StateTransferLock stateTransferLock, int topologyId) throws InterruptedException {
+         stateTransferLock.waitForTopology(topologyId, 1, TimeUnit.DAYS);
+      }
+
+      @Override
+      boolean isReady(StateTransferLock stateTransferLock, int topologyId) {
+         return true;
+      }
+   },
+   READY_TOPOLOGY {
```

Comment:
I feel we could merge the `WAIT_` and `READY_` variants, with only the cost of an extra volatile read for each command executed directly in the OOB thread pool.